Engine/Client/Src/Application/Application.cpp

```diff
@@ -52,17 +56,69 @@ void Application::_keyboardHandler(std::size_t key)
         // if (!_sceneManager->processInput(KEY_MAP(key), this->_client->getIdxPlayerComponent()))
         //     return;
         ABINetwork::sendPacketKey(_client, key);
+        // ABINetwork::sendPacketCreateRoom(_client, "BOB", false, "", false, 9);
     } catch (const std::exception& e) {
         std::cerr << "Exception: " << e.what() << std::endl;
     }
 }
 
-void Application::_connectServer()
+void Application::_packetHandler()
 {
-    if (!_client)
-        _client = ABINetwork::createClient("127.0.0.1", 4444, 4445);
+    std::lock_guard<std::mutex> lock(_client->getMutex());

```

Comment:
I think you can directly create a lock() method in the client class, so that you don't need to specify `std::lock_guard<std::mutex> lock(_client->getMutex());` but just writing `_client->lock();` (If you change it, you can remove the mutex getter)

Reply:
They're many places where you can replace the lock but an inner method

Reply:
I use it only one time, so I didn't see the point to create a method for ths.